### PR TITLE
ft(S3C-3754): Add bucketDeleteEncrytion route and support code

### PR DESCRIPTION
**Pull request by tmacro**

### CODE REVIEW

File: lib/auth/Vault.js [hunk 1]
```diff
@@ -222,6 +222,40 @@ class Vault {
...
+                    { errorMessage: err });
+                return callback(err);
+            }
+            const infoFromVault = info.message.body;
+            log.trace('info received from vault', { infoFromVault });
+            const result = {};
+            /* If the accountId was not found in Vault, do not
+            send the canonicalID back to the API */
+            Object.keys(infoFromVault).forEach(key => {
+                if (infoFromVault[key] !== 'NotFound' &&
```
Comment (jonathan-gramain) [inline lib/auth/Vault.js:250]: Do we need to hide those return codes? Thinking they might be used by the caller for error management.

File: tests/unit/auth/in_memory/backend.js [hunk 2]
```diff
@@ -16,6 +16,11 @@ const expectCanId =
 const searchEmail2 = 'sampleaccount4@sampling.com';
 const expectCanId2 = 'newCanId';
 
+const searchCanId = '79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7cd47ef2be';
+const expectAccountId = '123456789012';
+
+const invalidAccountId = 'doesnotexit';
```
Comment (jonathan-gramain) [inline tests/unit/auth/in_memory/backend.js]:
```suggestion
const invalidAccountId = 'doesnotexist';
```

**Review by jonathan-gramain — APPROVED** [review]

File: lib/auth/Vault.js (hunk 1 above)
Comment (tmacro) [reply lib/auth/Vault.js:250]: Stripping the `NotFound` keys is consistent with the `getEmailAddresses` and `getCanonicalIds` methods.

**Review by naren-rajendran — APPROVED** [review]

**Pull request merged**